---
title: "Chum GTseq Panel Evaluation"
output:
  html_document:
    df_print: paged
    code_folding: hide
    toc: true
    toc_float: true
    toc_collapsed: false
---

```{r, message=FALSE, warning=FALSE}
require(tidyverse)
require(DiagrammeR)
require(poppr)
require(genepop)
require(graph4lg)
require(related)
require(adegenet)
require(knitr)
require(insect)
```


# Readme

This is an rstudio project. If you'd like to pre-rendered figures, read a summary of analysis and view code, please open the relevant html file in a browser. 


To conduct the analyses on your computer, edit or run code: clone this repository into a directory on you r local machine and open the .Rproj file in Rstudio. All data (except raw sequencing data) and analyses are available in the github repository at https://github.com/david-dayan/Chum_test_panel



# Rationale 

In this notebook, we check if GTseq markers developed to genotype WA state Chum salmon populations will work well in coastal OR Chum populations. SFGL received an aliquot of primer mix from WDFW and ran these primers with 95 coastal OR Chum samples as a spike-in in a larger GTseq run.

# Goals

(1) Look for failed primers (extremely low reads across all samples)
(2) look for read hogs (over amplifying primers)  
(3) primer specificity  
(4) probe specificity  
(5) dimer formation  XXXXX did not do this XXXXXX

__Approach__  
This can mostly be accomplished by pooling all the reads from the test run (Oke samples) into a single fastq, hashing the file, then running some included optimization scripts from the gtseq pipeline.

(a) GTseq_SeqTest  
The GTseq_SeqTest script searches for unique (exact) occurences of each primer and  probe, and when both occur together. This can be used to identify low primer specificity (high ratio of probe:primer) or low probe specificity (high ratio of primer:probe). We can also look at the distribution of reads to identify any read hogs, or estimate if any poorly amplifying markers can be resuced with the increased depth from removing read hogs/reducing their primer concentration. I couldn't get this script to run correctly, after a few tries, so instead wrote a similar script with awk and seqkit.

(b) Genotyping
We can run the genotyping script and then look for paralogs, and adjust allele correction values or change in silico probe length to increase specificity to avoid paralogs. Similarly, we can look for loci with non-normal distributions of reads across individuals to attempt to identify if primer or probe sequences occur over common variants. We can "rescue" these markers by shortening the in-silico primer or probe sequences to avoid the variant, but then this new in silico primer/probe needs to be tested again for specificity.

(c) GTseq_Primer-Interaction-Test XXXXXX did not run this XXXXX
This script looks for the reverse complement of the reverse primer sequence in off target reads, thereby identifying reads that formed from primer-dimers. it runs off the hash file. See note in the section below for why I didn't run this.


# Data Summary

## Sequencing Data and Populations

Raw compressed reads are at /dfs/Omalley_Lab/dayan/coastal_chinook_2020/full_panel/demux/keta_reads . 

95 samples library from four populations spiked into a larger coastal chinook run. Demuxed with deML. The total number of reads was 50,680,864 .

Populations: Tillamook River, Nehalem, Mill Creek (Yaquina), and Yaquina (Mainstem)

```{r, eval = FALSE}
pops <- unique(str_sub(gt$Sample, 8, 11))
```


## Probe Seqs

We received primer-probe sequence information for two separate rounds of optimization; an early version and a version modified to accomodate 75bp reads. 

Let's check if the primer sequences sent along are the same.

```{r, warning=FALSE, message=FALSE}
primers1 <- read_tsv("panel_info/oke_panel_info.txt", col_names = FALSE)
primers2 <- read_csv("panel_info/Oke350_75c_GTseq_primerOrder.csv")
primers2 <- primers2[1:350,]
primers2$primer <- str_sub(primers2$Sequence, 33, nchar(primers2$Sequence))
primers2$marker <- str_sub(primers2$`Primer Name`, 1, nchar(primers2$`Primer Name`)-2) 

colnames(primers1) <- c("marker", "probe1", "probe2", "primer1")

primers1 <- primers1 %>%
  left_join(select(primers2, marker, primer))

primers1$same <- primers1$primer1==primers1$primer
```

It looks the updated primers include 5 changes. The data we have used an aliquot of primers from the former set of primers (primers1 in code chunk above), not the set from the 75bp optimized panel. 


# Data prep

First step is to decompress and move files to a working directory. 

```{bash, eval = FALSE}
#!/bin/bash
#$ -S /bin/bash
#$ -t 1-96
#$ -tc 30
#$ -N decompress_fastqs
#$ -cwd
#$ -o $JOB_NAME_$TASK_ID.out
#$ -e $JOB_NAME_$TASK_ID.err

FASTQS=(`ls -1 *r1.fq.gz`)
INFILE=${FASTQS[$SGE_TASK_ID -1]}

gunzip -c $INFILE > /dfs/Omalley_Lab/dayan/chum_test_panel/genos/${INFILE%_r1.fq.gz}.fastq

#save this code chunk as a file on the server and submit this with qsub -q harold scriptname from the directory you want the output .genos files
```

Next we'll create a hash of all reads

```{bash, eval = FALSE}
# first merge all sample reads into a single file
cat * > merged.fasta

#run hash (did with interactive shell (qrsh))
perl /dfs/Omalley_Lab/dayan/software/GTseq-Pipeline/GTseq_HashSeqs.pl merged.fastq > merged_hash.txt

```


# GTseq_SeqTest


## Log 
This script (from the gtseq pipeline) performs and exact matching search among the hash of reads from the test panel for (1) primer sequences, (2) probe sequences and (3) both 1 and 2 together in the same read.

```{bash, eval = FALSE}

perl /dfs/Omalley_Lab/dayan/software/GTseq-Pipeline/GTseq_SeqTest.pl oke_panel_info3.txt ./genos/merged_hash.txt > test_outputs/seq_test2.txt

# heres an example script of doing the same thing manually
cat genos/merged_hash.txt | ../software/seqkit grep -s -i -p GCAGGACACAGCGTGCTA | awk 'BEGIN {FS = ";"} ; {sum += $3 } END {print sum}'
```

The seq_test script keeps putting out wrong values (and take 4 hours to run...) I think it has something to do with the way it is parsing primer and probe info from the input file. Rather than continuing to deboug either my inputs or the script, just wrote a simple awk + seqkit one-liner to find matching primer and probe sequences in the hash file. 

First some prep. Need to make a list of probe sequence, with reverse complements

```{r, warning=FALSE, message=FALSE}
panel_info <- readr::read_delim("panel_info/oke_panel_info2.txt", delim = " ", col_names = FALSE)
colnames(panel_info) <- c("marker", "primer", "probe1", "probe2")
panel_info$probe1rc <- rc(panel_info$probe1)
panel_info$probe2rc <- rc(panel_info$probe2)

#write_tsv(panel_info, file = "panel_info/panel_info_for_awk.txt")

```

Now we iterate through the list of probes, primer and both and populate a file of counts

```{bash, eval = FALSE}
#!/bin/bash
#$ -S /bin/bash
#$ -N primer_count
#$ -cwd
#$ -o $JOB_NAME.out
#$ -e $JOB_NAME.err

PRIMERS="CCCACCCTGGACAGAGATAAAG
GCAGGCTGTCTCGTCATTGTT
CTGTCAATGGAGGGTGATCGAAAT
AGAATGACTATCAACAGCCCTAGGT
GCACCCCACCTGGCTT
AGGCAGCCAGGTAAGTCAGTA
TCCTCCTGCAGCTGTAAGTT
TGCAGGAGAGCTACCAGACA
CCTACTATTTCAGAGGCTTGACACA
CAGGGAACAATGAGGGATTG
GGGTGGCTACGATGTACAGA
ACTGTAGGTCCCCGAGAGAAG
TGCAGGTGGATACCTCTGA
GGCAAAGTCGGGTTGAGAT
GGAGTTGGCGGTCAGACA
TCTCACAACAAAAACTGGCAAATCTG
AAGTGGATTCCATGCAACTG
ACGGTGTTTCCTCCACACAT
TCTTTTGTCAGTGGTGGATGA
GTGTGTTGGAAAGCTGCTCA
TCTTATCCCTCTTGTAATTCTTCC
GTACAGCAGAGACTAAAGCTATACAACA
GGGCAGTGTCCTACAAG
AATGCCGCCTCCACACA
CAGGCCGGAAGACTGAGA
TGCAGGACTTTGTTCCTCTT
CAGGTGCGTTTGACCACTG
GCACCTCCCAATGGCTTTCT
TGGAAACCCTGGTCTAGTGG
CAGGGTGGATGGCTCCAG
GCAGGACACAGCGTGCTA
AAATTCCATGTGTGAGGAACA
GAGGCTTCAGTCGTCCTGTATC
TTCCTTGCTGTCATCTCTGC
TAAGAAGCCCAAGGCAGGAG
TGCAGGTCCTGGTATACATCTG
TGCTGGATGTGCTCAGACAA
CATCCCAGAGGAACAGAGGA
ATGGAGAGTGGGTGCAGAAA
TGCAGGTTTTAACTCTCACGTT
TGGGATCTGACTGGAACCTC
TGCAGGCTTCAAGTAAAGTGTT
TGCAGGGGTCTGTGTCTATG
ATTCCAGCCAGTGGTTGTGT
GGGAAGAAAAGCCTACCATAAACAG
GTGCTCTATTTCTGAGTTGGTGTGA
GCTCGACCTTATTCATGGTTTCCT
CGCACCATTGCTAACAAAAC
GTCTCAGGGCCACCTTTTT
GTTCGTTGCAATCGAGTCAATAGTG
TTCCAGTCCTGGTGTGGTCT
GGCTGGTAGATTTCCTGAAGC
TCTGGGCCGATTAATCATTG
TGGCAATGGGATGTCAAGTTATGA
CAGGAGGAAGGCATTATCGTTAGTG
GCAGGGTAGATGCATGATGA
GCAGGAAAGGGATCTGGAG
AAGTGTCCGTCTCATTACATTGTC
GTATGTGGTGCGGAGCATC
CACCCCATGTGCCTTTGGA
AGAGGGATTGAGTGCTGCAT
GCAGGTCCAGAACGTCCA
GAAATGCAGCGAACAGAGC
GCTTAACTGCTGTTTCTGCTATAGG
CGTCATCAGATTTGAAAAGGCTTTGC
TTGGTTATTTGCGGGATTGTCATTG
TGCAGGCAGACTATAATGTCATCAA
CACAATGCACTTTAGGCTTGA
CATTACCAGGGTCCATCACC
ACGGCGAGGGCAGGTATT
CATGGCCAAAGTCAGGTACA
CCAACACCACTTTCCATTAATAAGCA
CCAGACCATGTGCTTGTTTGTCATA
CGCTCCGTCCAGTGATGTC
CGTCGTGAACCATGAGTGCAATA
GCGGCTGACGATGAGGAA
ACACAGTGAACAATGAGGACTACAC
CGGAACCGATCTGAGTCTGTTG
TGACTCGTGTTCAGGAGAGG
GCGTCCACGCTCGTCAT
GGCCCCCAACCATCCT
GCAGGCAAATATTACCCCAATCCTT
TGCAGGGTTAACATCCTGTACT
ACGACGCCAGTAGCTACAAA
GTCACCAAACATCCTGCGAATG
CCCGGTTCCTCCAGTAGTTG
CGTGACAGCAGGGAGATGA
TGCAGGTTAAAATGGCGAATGTG
GGTCACGCCAAGGTTCTTT
AGCTTGGCTGTGGTGTTTCT
GCAGGTGTGATCTTTGGATG
GCTAGCACTTCATGGCAGCAT
CTTCCCACGTCTTTTCTGTCTCA
GTCTGTATACCAAATAGTGATGAGTGCAT
CCCTGTGGGTTCATACTGTAACT
GGCATGCGAATGTTAACCTGTTTA
CCTTAAAGTTGACCCCCAAA
AGGAGTCCAAAGCCCACAG
CTGGGAATCGACACAAATGA
CAGGTGACCTTTGCCTTGTA
CCCTGCCATTTAGTACAGGTC
GCTGCTGTCTTAAACCACATTCTACA
GCTCTGGTCCAGGTCGTT
CGCCAATAACGCTCCAACAAC
CCTGTACCTATAGGGCAACTTCAC
GGCAGGATAAGCAATGGAATGC
CACACAGAGCACGTGCAGAT
GCAGGTAACCCTGCCTTCTA
CAGCCACTGTAGCAGGGAAT
GCCTGCAAGCAGTAAACACA
GCCAATCGCGGAAGTCTCA
GCACTGCTCATTCTGTCACATG
TGAGCAGCCAAGCAGTTAAA
ACATGTTGCTCTTGGCTCCT
GGTGATTGGATGGAGGCTAA
TTCACCAGAGTCCCTCCACT
GTCAGACAGCCAGGGAAGAC
CAGGTGCAGTCAGCTTGTTC
AGGCTGCTTTAGAGCCACAC
CCCAGAGGCCTCGATTATCT
GCAATCCTCCATACATTACCTGTCA
CACTCCCCTCGCTTACTGATATCTA
GGCTTGAGGGCCACATTG
CCATTTCAGCATTGCCAGATTTGAAA
ATGCAGGTCAACTGAGGTAAA
GGCCTCCACATCGACTTCTC
AGGCTGTGTCATCGTTGTTAGTAAT
GCAGTTCAGTTCAGTGATTGCCTAA
ATTGGAGGGAACAGCACACT
AGGGATGGGGGTTAGGATTA
GCAGGAAATGCAAATGAGC
TTGAGGAAACCACTGGTCTTACAAG
TGGTCATAGCTTGCACTGTACAAA
GCACCCAACAGTTCATCATGTAAGT
CCAGCCTATTTCCTTTAGTGCATATGA
CGGTTGTCAGTTGAACAGTGTTT
GCCGCACCTCACACTAGT
GCTCCTCTCTGCCTGGTT
CAGTTGTAACTAACCTGGCTCTGAT
ATCGCCATCACGTCTCTCTC
CCAGTGCTTTTGTTCCCATC
GCAGAACTCGAGTCCAAAGG
GCAGTCACACATTTTCTTCATCACA
TCTTAAAATGGAGAGAGCGATTAATGAAGG
TGTCCTAATGACAGGCCTTGC
ACGCCCTTCCCCGATTC
TGATGAAACTGTGGCCTTGCA
CAGGAGGAAGGGCAATGTT
GAGTTGGCGTTGGCTGAC
CGCTACATGCAGGGTGTCT
TCAGGGTGAACTGTCCCTTT
CCCATAATTTTGCAACCCTAGTCACA
GACCCCGGGTCACCAC
CTGTGGATGGTATTCTGGATGCA
TGTCTTCAGAATCCAATGCTTTCCT
AGGAAAATAAACACACAGCACACCTA
TGCAGGTTCAACAGGG
TGCAGGTAACATCAGAGCACAAA
CAAAAATCACCCAGGAATGG
TTTTGGTAGGCCGTCATTGT
GGATGGTGACACACTTGCTG
GGAATCCTGGAAGAAGGCTAA
TGGGAGAAGACACAACCACA
TGTCTTCATCTGTGGACTCCTACA
AGGTTTGTATGCGGCTGCTT
CGTCTTTCCAATACCACAGAGATACA
AAATCCCCATGGAGAAACACAATGA
CTTAACTCAGTCACACCAACTCACT
GTGCACACCAATCCATTTCTGAAT
CTCAAATTTCCCTTTGACATCAATTCATCA
TTCTAGCCATGACCCAATGAAAGG
CCAGGACGAAGGTCAAAGTTCTT
CCCCTGGGAACAAATGAAACAGAAT
CCACATATACCGTGCCTTCAGAAAA
CAGTCCGCCACAATGAGTTG
GAGTTGAATGCAGGGAGAATAGGT
GGATGACTAAGAGCGACTGTATGTG
CGACGTGCAGATGTTTCATT
TGCAGGAGGTGAAGCCTCT
GGTGAAGGCACATAAGGTGTG
GGGTGGAGTATGCCCAGAAT
GGAAGGGTGCTGTGGAGAGT
CAAGCAGGCAGGATACTGGT
TCAGCAGAAGCCAAGTCAGA
TGCAGGCCCACAAGCTG
GAGCACATCACAAATGCAGAA
TCTGAGAGGCTCTGGAGGAG
TCCACAGTGAGTAATAAAGTTGCACAT
AGTTGGGTCTTAAAGATGATCATTTGCT
TCAGGGACGATAAAGGGATCATCTT
GGGCCATTCTCTGAGTTCAGT
CACCGCACATGGCAGAAC
GGGCATTTCATTTAAACCCAAACC
GCTATCAACGCACAACGTGTT
TAGGGTGTCGATACGGGTGT
CCCCAAATTTTCAGAGCAAA
TGCAGGCAGTCACTAAGTCTG
CAATCTGCAGGAGTTCCACA
TATGGGTGAATGGGGCTAGA
ACGGACGTCTTAGCGCATAC
ACCTGCTCAGCTCTGCTCTG
CCCCACTCCTTCTACTCATCCAT
CATTGTAGCAGAGGGTCAACGATAT
TCCACGTCACTCAGCATGATG
ACATTGTAATCTGTATTAGTCCAATGCAGAC
TGCAGATCTCAATTTGAACGACCTAT
ACTGAGGCAATATTCTGCAGGTT
GGGTGATGGTGGTGCCA
TTCGATAAGCTAGACCTGTGGGA
GCTGAGGGCTTCTGTTGCTAATTA
CTGGCAGGGTGTTGATCCA
CAATCTGACCTAGTCTTCCAAGCC
GGTGGACTTTGGCAAACTGTTT
TGCAGGTTAGCTTGGTTGGTTAA
CCACAGAAGCCTACGTACCTG
CAGGCTTGTCCCATTGCT
TCGAGGATTTGAGGATTAGGCTACT
GGTTTGATTTCGTCGCGATTTGA
TCTGCTTTTGTCGTCTCACCAA
CCATCACGTCACCATTACACTGT
GACACAACGTTTTGGGACATTCAG
GCAGATATGCCTCAGGGATGT
CGAAAGAAGATGGCTGCTGTGA
TGGTGTTCTCTATTGGCATGTGTT
GCAGGAATAGAATTTACCCCACCTC
CCCAGAGCAGACTGGGTCTC
AGTTTTGTTGCACCTGGACA
CAGGGAAATGAGAAGCCTGA
ATGCGGGAATGAGCAGATAA
GCCAAGCCCAGACTACACAT
TGCAGGTCAGCAAAATAATCT
GCAGGGTTTCAGCTTCAGTT
TCCCAGTGGTAGAGTGTCACC
AGAGCAATTTTCCCGGTTGT
GGTCCTACCTGCAACCTGAG
CTAAAGGCCCGCTCTGGAC
GTGCACCATGACAGTGAAGC
TGGTTACCCACTGAGCTGTTC
CCACAACAACATCTCCCAAA
TGCAGGTTTTGTCAGTGGC
TGGCCGATCTCTTCATGGTAATC
GAAGACTGGACACACACGATATGT
GCATCATTAGTAAGGCGTCATTGG
CAGAAGGTCTAAAGGTCCTTAACAATCA
AGGGCAACTACGCTCATAGAGTA
ACTGTGACTGAGTGTGCAATGTTAT
GGCCCGGCCTTGCT
CGTGACCTGTTACTCTGCTATCTTT
GCAGAGAAGCAAAATGGGTAGTTG
TGCTCGGGCTCCAAATGAC
TGCAGGAGGAACCGCC
TCCCCATCCGTCTCTATCAC
CACTCACTAGTGCTCCCAGAG
TGCAGGATCGTCTGAGGAG
GCCCGATGATGCCTATATCGT
AAAGACAGAGATTGGCTGAGG
GATTATTGGATCGAGAGAGTTAGAAGAGAGA
CCTAGACCACTCCAGACTGTTG
CCGCAGACAGTGGTCAATACT
GAATTTGACTGCCTGGCGAAAG
CATATCAGGTGTGTCTCAACAGTCT
GCTACCTCTATCATACCGCCAATATT
GGGAATAACAGTCATTGCAGCATTT
CAGGCTCTTCCTACTGAGTGAATG
CTCATTGGATATCTCATTGGCAGACA
ATGAGACTATAAATCACATTGGCGACAT
GCAGGAGTGAGTCACTGCTTTA
GACCAAATCCTTCACGTCACTGA
AACACCTCCTCCCAGAAAGG
GGGTCCTGATCCAACTGTGT
ACATTTGGTGGTGCTTCAGA
TGCAGGAGAGAACATTCTGG
GGAACAATGAGCTGGAGAGG
TGCAGGTAATTAGAGGCCGAG
GGTGATGTCAGTTGCTCTCG
CAGGATCGTCACCTGGACA
GGACGGCTTTGTTGACCA
CCAAGAGGAAGCAGCAGGAT
CATGCTGGTGAATTATTGGACAATGT
ACGTGTCCTGTTCAAATTAGCAGTA
GCAGTTGTCATGGCAGTAAGG
GCTTCTATGGGTCCTACTACCGTAT
CCGATGACTCCAAAGAAGTTGCT
AACCACAAGGAGAGAGGAGAGT
ACTTCTGGAGTGTGGCATCAAAA
GGAGAGGCAGACTGTAATGACA
TGCAGGAATGTGCCGGG
TGCCTGCTTTTCCACACATA
AGGCATCCAAACATGAGAGG
GCCGAAAAGATTTGGCACT
GGTTGTTCCGTTCCATTGGTG
AGGGAAGACCTGAGGTTGG
TGTTGGGGAATAGAGGCTGT
CTCCAGGGCATCAGCAAATG
CTCAAGCCCTCCACACTCA
GAACAGCAGTGAATCGGTTACCT
AATTGCTCCTCCTCGCTATTTCTC
AGGCTGAGGCAGTGACATG
GGATGGACATTCACAGTCTGGTT
TGCCTACGATGGACCCAGAA
ACACAACTGACCATATTGACTTTCGA
CAGGAAAATTCACAATTACAATAGAGCTTGT
ATGCAGGATATCTGCTCCGG
GGCTGATTTGGTAGCCCAAGAA
TGGTCCTACAGACACAAAGAAACAA
CAGAGATGCGTCCGTGGT
CAGGATGAATGAATCCTCCAC
TTGCTTTGTGGAGTTTGGTG
CCCACACAGACTGAACACCA
CCGTTTCTGTCAGACTCTGGTAAA
CAGGAGCCATTGGAAGAGTAGAG
GCTATTCCAATGTAAATCCTGTACTGTGT
CACAAACCTACTACAGACAGCAGTT
CGCAATATTAAGTGCGTCATACTTGT
GCGGACACCCTCTCACA
TGCAGAAACTGATCATTGGTAGTGG
TCTGTCAACAACCACCAGAATAGTG
TTGTCCAACAGGGCTTGTCT
CCAGGTCTTCCGTGCTGAAT
CTTGATTTAGCTTAATTGTCTGGTGTGC
CTCCTCTCACGGCTCACTTTT
GCAGGACGATGGCAGATACAG
CCCCAGCAAATGACTTTAGGTGTAT
TCAGGAGAGTCATGGGTGGT
GACAATGTGTTAGTGTAGGCTTCACT
CAGAGTGCAGAGTAATACGCATACA
CTCCTCACTAACAGTAGCTGCAATT
CACACTTTCAATCAAATGTTGTGCAG
TGAGTCCATGTACTACAAGAAGATGCT
GCAGCAGACACACCGAAGTA
GTCACCCAGAACAATAGAATGAGTCT
ACTGTAGTGACTGCATTTCATATTGCT
GCACCATACAGCTCCACACAA
AGAGGCACACCCAAGGAGT
GACCAGGCCGTGTTTGAC
CGTCTCATTCAGCTCTTTGATGTC
TTTGGAAACGTGTGGGATTAATTTTGT
CGTGGTTGGTTTCATTGACTCTCA
TGCAGGGAAGTGTGGG
CAGGTAACTGTGCATCACTGC
AACCGCATGCTTACTTCACC
GCCACAGCACAGACAAACAA
ACCTTCCAGCAAGACAGGTA
CCGTCATAAAACAGTGGGAGA
AGGATTGTAGCGGCAGAGAG
GTCCTCCCTATTCTTCCACTTACCT
CTGAAACTACCTCCCCCTAACAAG
CCGTGGGTTCACAGAGTACTG
GAAGGCTCTCCCCCTTTCTG
CCCTTACCACCGCTAACAAC"

for i in $PRIMERS
do
cat /dfs/Omalley_Lab/dayan/chum_test_panel/genos/merged_hash.txt | ../software/seqkit grep -s -i -p $i | grep ">" | awk -F ";" '{sum += $3} END {print sum}' >> /dfs/Omalley_Lab/dayan/chum_test_panel/test_outputs/primer_counts.txt
done


#save this file as a script and submit with qsub
```

```{bash, eval = FALSE}
#!/bin/bash
#$ -S /bin/bash
#$ -N probe_count
#$ -cwd
#$ -o $JOB_NAME.out
#$ -e $JOB_NAME.err

PROBES="probe1,probe2,probe1rc,probe2rc
CTAGTTCCTCTCTACTTCCA,CTAGTTCCTCTCTTCTTCCA,TGGAAGTAGAGAGGAACTAG,TGGAAGAAGAGAGGAACTAG
CAACAGTAGTAGGCTTGATTACC,CAACAGTAGTAGGCTCGATTACC,GGTAATCAAGCCTACTACTGTTG,GGTAATCGAGCCTACTACTGTTG
TGTGCTACTATGTTGTCGTTC,TGCTACTATGTTGTCATTC,GAACGACAACATAGTAGCACA,GAATGACAACATAGTAGCA
CTTCTGGCAGTGTTCATA,CTGGCAGCGTTCATA,TATGAACACTGCCAGAAG,TATGAACGCTGCCAG
CCTCTCTGCAGGGCT,CCCTCTCTACAGGGCT,AGCCCTGCAGAGAGG,AGCCCTGTAGAGAGGG
ACAGAACAGAAACAGTG,AACAGAACAGTAACAGTG,CACTGTTTCTGTTCTGT,CACTGTTACTGTTCTGTT
AATCACCTTGT,AATCATCTTGT,ACAAGGTGATT,ACAAGATGATT
CACAGGTACTACTGACTGACTGTGGAAGGGAGCAGCAATGGAGCGTGACAG,CACAGGTACTACTGACTGACTGTGGGAGGGAGCAGCAATGGAGCGTGACAG,CTGTCACGCTCCATTGCTGCTCCCTTCCACAGTCAGTCAGTAGTACCTGTG,CTGTCACGCTCCATTGCTGCTCCCTCCCACAGTCAGTCAGTAGTACCTGTG
ATAAATTGTGAAGCAGAGGAGTTGAC,ATAAATTGTGAAGCCGAGGAGTTGAC,GTCAACTCCTCTGCTTCACAATTTAT,GTCAACTCCTCGGCTTCACAATTTAT
GTTTTCAAGCT,GTTTTTAAGCT,AGCTTGAAAAC,AGCTTAAAAAC
ACAGAAACTCA,ACAGAGACTCA,TGAGTTTCTGT,TGAGTCTCTGT
ATGTCACTGGGCTAAC,ATGTCACTGAGCTAAC,GTTAGCCCAGTGACAT,GTTAGCTCAGTGACAT
CCTCCCAAATC,CCTCCTAAATC,GATTTGGGAGG,GATTTAGGAGG
GTCATCACAAC,GTCATTACAAC,GTTGTGATGAC,GTTGTAATGAC
ACAGACGACTT,ACAGATGACTT,AAGTCGTCTGT,AAGTCATCTGT
CAGTGCATGTCCTCCTCGTGG,CAGTGCATCTCCTCCTCGTGG,CCACGAGGAGGACATGCACTG,CCACGAGGAGGAGATGCACTG
AGGGCAGTGTT,AGGGCGGTGTT,AACACTGCCCT,AACACCGCCCT
CCACACATTGGTACTTCTGGGTTAATT,CCACACATTGGTGCTTCTGGGTTAATT,AATTAACCCAGAAGTACCAATGTGTGG,AATTAACCCAGAAGCACCAATGTGTGG
TCTCTATGTCT,TCTCTTTGTCT,AGACATAGAGA,AGACAAAGAGA
TGTGGAGCTAG,TGTGGCGCTAG,CTAGCTCCACA,CTAGCGCCACA
TCCCAATTCCT,TCCCACTTCCT,AGGAATTGGGA,AGGAAGTGGGA
TGACTGACTGCATAGTTGT,ACTGACTGCGTAGTTGT,ACAACTATGCAGTCAGTCA,ACAACTACGCAGTCAGT
AAGACCATCACTTTTGCCTG,ACCATCACTATTGCCTG,CAGGCAAAAGTGATGGTCTT,CAGGCAATAGTGATGGT
CCCCTCTGACTTTTATCAA,CCCTCTGACTTGTATCAA,TTGATAAAAGTCAGAGGGG,TTGATACAAGTCAGAGGG
TGAGAGACAAA,TGAGATACAAA,TTTGTCTCTCA,TTTGTATCTCA
CAAGACGCTTT,CAAGATGCTTT,AAAGCGTCTTG,AAAGCATCTTG
GCAGTAGCCCC,GCAGTGGCCCC,GGGGCTACTGC,GGGGCCACTGC
CATCAGTGCCACCCCCA,TCAGTGCCGCCCCCA,TGGGGGTGGCACTGATG,TGGGGGCGGCACTGA
ACCTGATGAGA,ACCTGGTGAGA,TCTCATCAGGT,TCTCACCAGGT
CAAGGCCCCTG,CAAGGTCCCTG,CAGGGGCCTTG,CAGGGACCTTG
ACAGCGTGCTACGCTGAACAGCTGGAG,ACAGCGTGCTATGCTGAACAGCTGGAG,CTCCAGCTGTTCAGCGTAGCACGCTGT,CTCCAGCTGTTCAGCATAGCACGCTGT
AACATACTTTT,AACATGCTTTT,AAAAGTATGTT,AAAAGCATGTT
CAGCATCACGTCTCCCA,CAGCATCACATCTCCCA,TGGGAGACGTGATGCTG,TGGGAGATGTGATGCTG
CTGCAAAGTGC,CTGCAGAGTGC,GCACTTTGCAG,GCACTCTGCAG
GCCGGGGTGTC,GCCGGTGTGTC,GACACCCCGGC,GACACACCGGC
ATGTGGCTGTCCTTC,ATGTGGCTCTCCTTC,GAAGGACAGCCACAT,GAAGGAGAGCCACAT
TTGACCGCTCTTTCTGT,TTGACCGCTTTTTCTGT,ACAGAAAGAGCGGTCAA,ACAGAAAAAGCGGTCAA
TAGCTAGCTAA,TAGCTTGCTAA,TTAGCTAGCTA,TTAGCAAGCTA
AAGGCCGCTC,AAGGCTGCTC,GAGCGGCCTT,GAGCAGCCTT
CGTTACGGGGT,CGTTAGGGGGT,ACCCCGTAACG,ACCCCCTAACG
CTGGGATCTCA,CTGGGCTCTCA,TGAGATCCCAG,TGAGAGCCCAG
TAGCTACCTTC,TAGCTGCCTTC,GAAGGTAGCTA,GAAGGCAGCTA
AAAAAATATTT,AAAAAGTATTT,AAATATTTTTT,AAATACTTTTT
TTCCCAGAAGT,TTCCCGGAAGT,ACTTCTGGGAA,ACTTCCGGGAA
ATGTCAAATCACGCTGATG,ATGTCAAATCACACTGATG,CATCAGCGTGATTTGACAT,CATCAGTGTGATTTGACAT
CCTGACCCCAACCAA,CCTGACCCAAACCAA,TTGGTTGGGGTCAGG,TTGGTTTGGGTCAGG
TGAAGGTGGTGGAATTA,TGAAGGTGTTGGAATTA,TAATTCCACCACCTTCA,TAATTCCAACACCTTCA
CAACACACGCA,CAACAGACGCA,TGCGTGTGTTG,TGCGTCTGTTG
CGACTCGTCAA,CGACTGGTCAA,TTGACGAGTCG,TTGACCAGTCG
CTGAAATACACAATTTCT,CTGAAATACACCATTTCT,AGAAATTGTGTATTTCAG,AGAAATGGTGTATTTCAG
TTCTACACATT,TTCTATACATT,AATGTGTAGAA,AATGTATAGAA
CAGTCTGTCAAGGGGACCGTAAGGGGAG,CAGTCTGTCAAGGGGAGCGTAAGGGGAG,CTCCCCTTACGGTCCCCTTGACAGACTG,CTCCCCTTACGCTCCCCTTGACAGACTG
ATGGGCTGACA,ATGGGTTGACA,TGTCAGCCCAT,TGTCAACCCAT
AGAGAGTTGTCGTTCATC,AGAGAGTTGTCATTCATC,GATGAACGACAACTCTCT,GATGAATGACAACTCTCT
CTGCCTGTATAACCAAG,CTGCCTGTATGACCAAG,CTTGGTTATACAGGCAG,CTTGGTCATACAGGCAG
GGATTGACCTC,GGATTTACCTC,GAGGTCAATCC,GAGGTAAATCC
CTGCTGGGATG,CTGCTTGGATG,CATCCCAGCAG,CATCCAAGCAG
GAAACATTGCT,GAAACGTTGCT,AGCAATGTTTC,AGCAACGTTTC
GCATCCGGGCT,GCATCTGGGCT,AGCCCGGATGC,AGCCCAGATGC
CTAGGACCGATGTCCCT,TAGGACCGACGTCCCT,AGGGACATCGGTCCTAG,AGGGACGTCGGTCCTA
TTGAGTGCTGCATCATGACCGGGCCT,TTGAGTGCTGCATCATGACCTGGCCT,AGGCCCGGTCATGATGCAGCACTCAA,AGGCCAGGTCATGATGCAGCACTCAA
TCCAGCAGGTC,TCCAGGAGGTC,GACCTGCTGGA,GACCTCCTGGA
GCAAAACCCAA,GCAAACCCCAA,TTGGGTTTTGC,TTGGGGTTTGC
CATGATGGCGTCAAAC,ATGATGGCATCAAAC,GTTTGACGCCATCATG,GTTTGATGCCATCAT
CACCGAAATAGAATTT,CCGAAAGAGAATTT,AAATTCTATTTCGGTG,AAATTCTCTTTCGG
CATGCTTTAGACTATTTGAAT,ATGCTTTAGACTGTTTGAAT,ATTCAAATAGTCTAAAGCATG,ATTCAAACAGTCTAAAGCAT
AGAGTGGAGCATTCAG,AGAGTGGAGCGTTCAG,CTGAATGCTCCACTCT,CTGAACGCTCCACTCT
AAAGGCTTGCT,AAAGGTTTGCT,AGCAAGCCTTT,AGCAAACCTTT
TCACCATGTGTGGTGGAGGATAGGATCG,TCACCGTGTGTGGTGGAGGATAGGATCG,CGATCCTATCCTCCACCACACATGGTGA,CGATCCTATCCTCCACCACACACGGTGA
CGGCAAGTTGT,CGGCAGGTTGT,ACAACTTGCCG,ACAACCTGCCG
GCTCAAGGGAT,GCTCAGGGGAT,ATCCCTTGAGC,ATCCCCTGAGC
AAACAAAGGCAAAGTC,AAAACAAAGGAAAAGTC,GACTTTGCCTTTGTTT,GACTTTTCCTTTGTTTT
CAGATCCGGTGTATGC,ACAGATCCTGTGTATGC,GCATACACCGGATCTG,GCATACACAGGATCTGT
TGGGCATTTAGAGTTTATT,TGGGCATTTAGAATTTATT,AATAAACTCTAAATGCCCA,AATAAATTCTAAATGCCCA
CTCCCTCGTGTCGAGCA,CTCCCTCGTATCGAGCA,TGCTCGACACGAGGGAG,TGCTCGATACGAGGGAG
CTGCCCTGGCCGTATG,CCTGCCCTGTCCGTATG,CATACGGCCAGGGCAG,CATACGGACAGGGCAGG
TAGAATCCCAGCAGCTCT,TAGAATCCCACCAGCTCT,AGAGCTGCTGGGATTCTA,AGAGCTGGTGGGATTCTA
ACTGGCCATGACTGAC,TGGCCACGACTGAC,GTCAGTCATGGCCAGT,GTCAGTCGTGGCCA
AAGAGCCAGTG,AAGAGTCAGTG,CACTGGCTCTT,CACTGACTCTT
TTGAGGAGGGATTTAA,TTGAGGAAGGATTTAA,TTAAATCCCTCCTCAA,TTAAATCCTTCCTCAA
CTGCTTTTCAGTCTCTCCAA,TGCTTTTCAGTCCCTCCAA,TTGGAGAGACTGAAAAGCAG,TTGGAGGGACTGAAAAGCA
TGTTTCCTCCATTTAAGATG,TTCCTCCATTAAAGATG,CATCTTAAATGGAGGAAACA,CATCTTTAATGGAGGAA
CAGGTATAAAA,CAGGTTTAAAA,TTTTATACCTG,TTTTAAACCTG
ACAAACACAAA,ACAAATACAAA,TTTGTGTTTGT,TTTGTATTTGT
CCGTTCTCTTCTTGGACAC,CGTTCTCTTCCTGGACAC,GTGTCCAAGAAGAGAACGG,GTGTCCAGGAAGAGAACG
ACTACGAGAAGAATCCCAAAG,ACGAGAAGAACCCCAAAG,CTTTGGGATTCTTCTCGTAGT,CTTTGGGGTTCTTCTCGT
TGGCCTTCCAGCTGAA,TGGCCTTCCAACTGAA,TTCAGCTGGAAGGCCA,TTCAGTTGGAAGGCCA
ACGCCAGGCCAGGAC,CGCCAGCCCAGGAC,GTCCTGGCCTGGCGT,GTCCTGGGCTGGCG
TGCGCACTGGG,TGCGCTCTGGG,CCCAGTGCGCA,CCCAGAGCGCA
TGGATACTCTG,TGGATGCTCTG,CAGAGTATCCA,CAGAGCATCCA
CCGATCCTGTGCAGGTGTTGATC,CCGATGCTGTGCAGGTGTTGATC,GATCAACACCTGCACAGGATCGG,GATCAACACCTGCACAGCATCGG
ACGTTAGTTAGCTATGTCTG,ACGTTAGTTACCTATGTCTG,CAGACATAGCTAACTAACGT,CAGACATAGGTAACTAACGT
CAGATCAAATTGTAAAACT,CAGATCAAATTTTAAAACT,AGTTTTACAATTTGATCTG,AGTTTTAAAATTTGATCTG
AAGGCGTGCATAGGAAA,AAGGCGTGCGTAGGAAA,TTTCCTATGCACGCCTT,TTTCCTACGCACGCCTT
CTGTTTTGTACCATGACTTG,CTGTTTTGTACCCTGACTTG,CAAGTCATGGTACAAAACAG,CAAGTCAGGGTACAAAACAG
CCGACGTGATTAAGAC,CGACGTGAGTAAGAC,GTCTTAATCACGTCGG,GTCTTACTCACGTCG
AAAAAACATCTGGATGAGA,AAAAATCATCTGGATGAGA,TCTCATCCAGATGTTTTTT,TCTCATCCAGATGATTTTT
CCCATAGAGTA,CCCATCGAGTA,TACTCTATGGG,TACTCGATGGG
ACAAGAGATGT,ACAAGGGATGT,ACATCTCTTGT,ACATCCCTTGT
AACTCCGTTGC,AACTCTGTTGC,GCAACGGAGTT,GCAACAGAGTT
TCAGGATGTGT,TCAGGCTGTGT,ACACATCCTGA,ACACAGCCTGA
ATGGAATGACAGCAATGT,ATGGAATGACAACAATGT,ACATTGCTGTCATTCCAT,ACATTGTTGTCATTCCAT
TGGAGGAACGCGCGCC,TGGAGGAACACGCGCC,GGCGCGCGTTCCTCCA,GGCGCGTGTTCCTCCA
CACCAATAACGTCCTAATC,CACCAATAACATCCTAATC,GATTAGGACGTTATTGGTG,GATTAGGATGTTATTGGTG
CGGAGCCACTTCCAGTA,CGGAGCCACTACCAGTA,TACTGGAAGTGGCTCCG,TACTGGTAGTGGCTCCG
CTGATGAGACAGATCTTA,TGATGAGACAAATCTTA,TAAGATCTGTCTCATCAG,TAAGATTTGTCTCATCA
GTGCGCTTTAT,GTGCGTTTTAT,ATAAAGCGCAC,ATAAAACGCAC
TCGTTCGTGGT,TCGTTTGTGGT,ACCACGAACGA,ACCACAAACGA
GGGCTATAGAC,GGGCTTTAGAC,GTCTATAGCCC,GTCTAAAGCCC
TGCACATAC,TGCACGTAC,GTATGTGCA,GTACGTGCA
CTTCAACATCTCCTACCCTC,CAACATCTCGTACCCTC,GAGGGTAGGAGATGTTGAAG,GAGGGTACGAGATGTTG
CATCTTTGTATCTGTGCCATT,TCATCTTTGTATCTATGCCATT,AATGGCACAGATACAAAGATG,AATGGCATAGATACAAAGATGA
CGGAGCTGGCG,CGGAGTTGGCG,CGCCAGCTCCG,CGCCAACTCCG
ACCCTGCCGTC,ACCCTTCCGTC,GACGGCAGGGT,GACGGAAGGGT
CTGGACAGCCA,CTGGAGAGCCA,TGGCTGTCCAG,TGGCTCTCCAG
CACTACGTTTA,CACTATGTTTA,TAAACGTAGTG,TAAACATAGTG
ATATTATAACC,ATATTGTAACC,GGTTATAATAT,GGTTACAATAT
AGCCCAGTAATTAAGAGCATCTATCATCATTAGCA,AGCCCTGTAATTAAGAGCATCTATCATCATTAGCA,TGCTAATGATGATAGATGCTCTTAATTACTGGGCT,TGCTAATGATGATAGATGCTCTTAATTACAGGGCT
GTGAGAGACTA,GTGAGCGACTA,TAGTCTCTCAC,TAGTCGCTCAC
TGGTGGTGGCT,TGGTGTTGGCT,AGCCACCACCA,AGCCAACACCA
CAAATAAGACAGAAGCAGTCTGCCCCT,CAAATAAGACAGTAGCAGTCTGCCCCT,AGGGGCAGACTGCTTCTGTCTTATTTG,AGGGGCAGACTGCTACTGTCTTATTTG
CTAGATCATGTTCACTATCTATA,TAGATCATGTTCACTATATATA,TATAGATAGTGAACATGATCTAG,TATATATAGTGAACATGATCTA
TTGAATTTCGAAAATCC,TTGAATTTCAAAAATCC,GGATTTTCGAAATTCAA,GGATTTTTGAAATTCAA
CTGGTGATTGACGATCC,CTGGTGATTAACGATCC,GGATCGTCAATCACCAG,GGATCGTTAATCACCAG
CAGCCACCTCGATGAA,CAGCCACCTTGATGAA,TTCATCGAGGTGGCTG,TTCATCAAGGTGGCTG
CGGGACGCTCCATG,CCGGGACTCTCCATG,CATGGAGCGTCCCG,CATGGAGAGTCCCGG
AGTTGGCTGAAGACCCCATA,TTGGCTGAAGAACCCATA,TATGGGGTCTTCAGCCAACT,TATGGGTTCTTCAGCCAA
AAATTCAACACACCTGGTCTTCCAGCTC,AAATTCAACACACCTGTTCTTCCAGCTC,GAGCTGGAAGACCAGGTGTGTTGAATTT,GAGCTGGAAGAACAGGTGTGTTGAATTT
ACTCCCAATGT,ACTCCTAATGT,ACATTGGGAGT,ACATTAGGAGT
AACATGTTTAA,AACATTTTTAA,TTAAACATGTT,TTAAAAATGTT
GCAAATGAGCTTCGTGATTTACATACATTTACTGAAAA,GCAAATGAGCTTTGTGATTTACATACATTTACTGAAAA,TTTTCAGTAAATGTATGTAAATCACGAAGCTCATTTGC,TTTTCAGTAAATGTATGTAAATCACAAAGCTCATTTGC
ATATCTTTGAGACTAGATTAA,CTTTGAGACAAGATTAA,TTAATCTAGTCTCAAAGATAT,TTAATCTTGTCTCAAAG
CTGCTGTTCATAAAAGTA,CTGCTGTTCATACAAGTA,TACTTTTATGAACAGCAG,TACTTGTATGAACAGCAG
CTGAGATTCTTCATCTTTTAC,TGAGATTCTTCATATTTTAC,GTAAAAGATGAAGAATCTCAG,GTAAAATATGAAGAATCTCA
TTTAGTTCTAGTCAAAAGTAG,TAGTTCTAGTCCAAAGTAG,CTACTTTTGACTAGAACTAAA,CTACTTTGGACTAGAACTA
TCGAGATGTGTCAAAAG,TCGAGATGTATCAAAAG,CTTTTGACACATCTCGA,CTTTTGATACATCTCGA
TTGGTGACCAGTAATTT,TTGGTGACCAGTCATTT,AAATTACTGGTCACCAA,AAATGACTGGTCACCAA
TCCAACGTCTGCGCTAT,TTCCAACGTCTCCGCTAT,ATAGCGCAGACGTTGGA,ATAGCGGAGACGTTGGAA
TAGATTAGGCTCAGAGTTTA,ATTAGGCTCGGAGTTTA,TAAACTCTGAGCCTAATCTA,TAAACTCCGAGCCTAAT
CTGAAGTCCAT,CTGAATTCCAT,ATGGACTTCAG,ATGGAATTCAG
CATATCA,CGTATCA,TGATATG,TGATACG
AGCGACGCTCT,AGCGATGCTCT,AGAGCGTCGCT,AGAGCATCGCT
ACTGGTGAAGGGATATT,CTGGTGAAGAGATATT,AATATCCCTTCACCAGT,AATATCTCTTCACCAG
CATCAGGGAAAGCCTACAAA,AGGGAAAGCCGACAAA,TTTGTAGGCTTTCCCTGATG,TTTGTCGGCTTTCCCT
AGCTGGTTACGTGTCGTG,CTGGTTACGTCTCGTG,CACGACACGTAACCAGCT,CACGAGACGTAACCAG
CGACGTCAATGACTTTG,CGACGTCAATTACTTTG,CAAAGTCATTGACGTCG,CAAAGTAATTGACGTCG
CAATTCATATCTGAACCAAC,CAATTCATATCTGCACCAAC,GTTGGTTCAGATATGAATTG,GTTGGTGCAGATATGAATTG
TTTACATTAGT,TTTACGTTAGT,ACTAATGTAAA,ACTAACGTAAA
GTGTCAGATCA,GTGTCCGATCA,TGATCTGACAC,TGATCGGACAC
CCCTCGTCCGT,CCCTCTTCCGT,ACGGACGAGGG,ACGGAAGAGGG
TTTAAACTCCA,TTTAATCTCCA,TGGAGTTTAAA,TGGAGATTAAA
CATTATCTCCCTGAATGTA,CATTATCTCCATGAATGTA,TACATTCAGGGAGATAATG,TACATTCATGGAGATAATG
CCTGCTATCGTCCTCC,CTGCTATCATCCTCC,GGAGGACGATAGCAGG,GGAGGATGATAGCAG
ACATCTGCTGTCTTCAG,ACATCTGCTATCTTCAG,CTGAAGACAGCAGATGT,CTGAAGATAGCAGATGT
CCAGCCCCCTCTGAAA,AGCCCCCGCTGAAA,TTTCAGAGGGGGCTGG,TTTCAGCGGGGGCT
TGCACATTGTGACATTC,CTGCACATTGTTACATTC,GAATGTCACAATGTGCA,GAATGTAACAATGTGCAG
TTCTCAGACAACATCATC,TCAGACGACATCATC,GATGATGTTGTCTGAGAA,GATGATGTCGTCTGA
ATGAGAGAATATCGAAGAAT,ATGAGAGAATATCAAAGAAT,ATTCTTCGATATTCTCTCAT,ATTCTTTGATATTCTCTCAT
ATGGAAGAGAT,ATGGACGAGAT,ATCTCTTCCAT,ATCTCGTCCAT
TTTGAATTTGTTCTTAACTGACTTGCCTAGTCAAATAAAGGTTATATACAGTGCC,TTTGAATTTGTTCTTAACTGACTTGCCTAGTTAAATAAAGGTTATATACAGTGCC,GGCACTGTATATAACCTTTATTTGACTAGGCAAGTCAGTTAAGAACAAATTCAAA,GGCACTGTATATAACCTTTATTTAACTAGGCAAGTCAGTTAAGAACAAATTCAAA
ACCGTTGGTGGTGATGGCGGTGAT,ACCGTTGGTGGTGATGGCTGTGAT,ATCACCGCCATCACCACCAACGGT,ATCACAGCCATCACCACCAACGGT
GGAGGACAGGA,GGAGGGCAGGA,TCCTGTCCTCC,TCCTGCCCTCC
CACGTCAACGT,CACGTTAACGT,ACGTTGACGTG,ACGTTAACGTG
CCCAACGCGGTGATG,CCCAACGCAGTGATG,CATCACCGCGTTGGG,CATCACTGCGTTGGG
TCATGAAATGGGTTTCTAT,ATGAAATGGGATTCTAT,ATAGAAACCCATTTCATGA,ATAGAATCCCATTTCAT
TGTGCTAGGTGTTCTGG,TGTGCTAGGTATTCTGG,CCAGAACACCTAGCACA,CCAGAATACCTAGCACA
ACTTTTTTGTCGTTTTTTT,ACTTTTTTGTCATTTTTTT,AAAAAAACGACAAAAAAGT,AAAAAAATGACAAAAAAGT
TGGCTCAAACTTG,TTGGCTCGAACTTG,CAAGTTTGAGCCA,CAAGTTCGAGCCAA
AGTATATTGAGATGAATCCCAC,ATATTGAGATGAATGCCAC,GTGGGATTCATCTCAATATACT,GTGGCATTCATCTCAATAT
CAGGTGCGTGCAGTAA,TCAGGTGCATGCAGTAA,TTACTGCACGCACCTG,TTACTGCATGCACCTGA
CTAACTTTACAAAGACACTGC,AACTTTACAAAGGCACTGC,GCAGTGTCTTTGTAAAGTTAG,GCAGTGCCTTTGTAAAGTT
AGGGAGCTGCTGGACA,AGGGAGCTACTGGACA,TGTCCAGCAGCTCCCT,TGTCCAGTAGCTCCCT
TGTTTCTCTGTTGTTTTTG,TGTTTCTCTGTAGTTTTTG,CAAAAACAACAGAGAAACA,CAAAAACTACAGAGAAACA
CATACCCCTCGACTTAT,CATACCCCTTGACTTAT,ATAAGTCGAGGGGTATG,ATAAGTCAAGGGGTATG
TTGGCTCATTGCGCTATA,TTGGCTCATTGCCCTATA,TATAGCGCAATGAGCCAA,TATAGGGCAATGAGCCAA
CACCATGCCTGCGATAA,ACCATGCCAGCGATAA,TTATCGCAGGCATGGTG,TTATCGCTGGCATGGT
CACCAAGGTAAAAAT,CCAAGGGAAAAAT,ATTTTTACCTTGGTG,ATTTTTCCCTTGG
CGCAAATTCAT,CGCAATTTCAT,ATGAATTTGCG,ATGAAATTGCG
TGGTCAAGCAT,TGGTCCAGCAT,ATGCTTGACCA,ATGCTGGACCA
TGTCACCCCTT,TGTCATCCCTT,AAGGGGTGACA,AAGGGATGACA
TTGCCTCACT,TTGCGTCACT,AGTGAGGCAA,AGTGACGCAA
GTAATAAAGCC,GTAATTAAGCC,GGCTTTATTAC,GGCTTAATTAC
TAGGTCGACAT,TAGGTTGACAT,ATGTCGACCTA,ATGTCAACCTA
TTGTTGGAGGA,TTGTTTGAGGA,TCCTCCAACAA,TCCTCAAACAA
GCTGTCTGCCAAGGCTGCCATTGAGCATGCTCAGAAGATG,GCTGTGTGCCAAGGCTGCCATTGAGCATGCTCAGAAGATG,CATCTTCTGAGCATGCTCAATGGCAGCCTTGGCAGACAGC,CATCTTCTGAGCATGCTCAATGGCAGCCTTGGCACACAGC
CCGTTCCCTTT,CCGTTGCCTTT,AAAGGGAACGG,AAAGGCAACGG
CAGGCACACGC,CAGGCGCACGC,GCGTGTGCCTG,GCGTGCGCCTG
CAAGAACTGACCTTAGACAC,AAGAACTGACCTTTGACAC,GTGTCTAAGGTCAGTTCTTG,GTGTCAAAGGTCAGTTCTT
CAGCCTCATAAGATGTTTA,CAGCCTCATAAGCTGTTTA,TAAACATCTTATGAGGCTG,TAAACAGCTTATGAGGCTG
CTTCCGCTCTCTACTCC,TTCCGCTCTGTACTCC,GGAGTAGAGAGCGGAAG,GGAGTACAGAGCGGAA
AAATGAATGATAATGTGTTCTG,AAATGAATGATAATATGTTCTG,CAGAACACATTATCATTCATTT,CAGAACATATTATCATTCATTT
CTCCATCCCAAGATAGA,CTCCATCCCACGATAGA,TCTATCTTGGGATGGAG,TCTATCGTGGGATGGAG
CACACGCAACGTTCA,ACACGCGACGTTCA,TGAACGTTGCGTGTG,TGAACGTCGCGTGT
TCGTTCATACATTGCTTG,CGTTCATACACTGCTTG,CAAGCAATGTATGAACGA,CAAGCAGTGTATGAACG
TACTAAGTCTC,TACTATGTCTC,GAGACTTAGTA,GAGACATAGTA
AATATAAACGT,AATATGAACGT,ACGTTTATATT,ACGTTCATATT
AGTCTGAGGTG[CT]GAAAGGAGCTCCTTAAACTTGACCCCTCTTTAAG,AGTCTGAGGTG[CT]GAAAGGAGTTCCTTAAACTTGACCCCTCTTTAAG,CTTAAAGAGGGGTCAAGTTTAAGGAGCTCCTTTCAGCACCTCAGACT,CTTAAAGAGGGGTCAAGTTTAAGGAACTCCTTTCAGCACCTCAGACT
ATCAGGCTCCG,ATCAGTCTCCG,CGGAGCCTGAT,CGGAGACTGAT
TAGGCGGTATC,TAGGCTGTATC,GATACCGCCTA,GATACAGCCTA
TGCAGCGATAG,TGCAGTGATAG,CTATCGCTGCA,CTATCACTGCA
CTGGCCCAAAG,CTGGCTCAAAG,CTTTGGGCCAG,CTTTGAGCCAG
AGCCTTTCACATCCCCA,CAGCCTTTCATATCCCCA,TGGGGATGTGAAAGGCT,TGGGGATATGAAAGGCTG
AAACTAGTATTTCCCGTCTATG,ACTAGTATTTCCCATCTATG,CATAGACGGGAAATACTAGTTT,CATAGATGGGAAATACTAGT
ACGTTGCCTGTCCAC,ACGTTGCCTATCCAC,GTGGACAGGCAACGT,GTGGATAGGCAACGT
ATTTATGGGTGTTCCCC,TTATGGGAGTTCCCC,GGGGAACACCCATAAAT,GGGGAACTCCCATAA
CAAGAGGTAAGAAGCTAC,AAGAGGTAACAAGCTAC,GTAGCTTCTTACCTCTTG,GTAGCTTGTTACCTCTT
CAGGAAGTTCCCCGTGTTG,CAGGAAGTTCCCTGTGTTG,CAACACGGGGAACTTCCTG,CAACACAGGGAACTTCCTG
AATAGCCCTGTCCTTCAA,TAGCCCTGACCTTCAA,TTGAAGGACAGGGCTATT,TTGAAGGTCAGGGCTA
CAGGCAATTATAAGCTATCCCA,AGGCAATTATAAGCCATCCCA,TGGGATAGCTTATAATTGCCTG,TGGGATGGCTTATAATTGCCT
TTGTCCCCCAGGAATG,TTGTCCCCCACGAATG,CATTCCTGGGGGACAA,CATTCGTGGGGGACAA
CCTCTCAGCTGGGCTG,CCTCTCAGCTAGGCTG,CAGCCCAGCTGAGAGG,CAGCCTAGCTGAGAGG
TGGGAGATTTTATTATGGAACTT,TGGGAGATTTTATTATTGAACTT,AAGTTCCATAATAAAATCTCCCA,AAGTTCAATAATAAAATCTCCCA
AGGAAGCAAATGACCCCT,AGGAAGCAAATAACCCCT,AGGGGTCATTTGCTTCCT,AGGGGTTATTTGCTTCCT
CTTCCGTTCAAATCTAAAA,TCCGTTCAAAGCTAAAA,TTTTAGATTTGAACGGAAG,TTTTAGCTTTGAACGGA
TGTGGCTTCTT,TGTGGTTTCTT,AAGAAGCCACA,AAGAAACCACA
AACGTCTAAGG,AACGTTTAAGG,CCTTAGACGTT,CCTTAAACGTT
TGTTTCCACAAGAACTGA,TGTTTCCACAATAACTGA,TCAGTTCTTGTGGAAACA,TCAGTTATTGTGGAAACA
AGGTGTACTGAAGAGAC,AGGTGTACTAAAGAGAC,GTCTCTTCAGTACACCT,GTCTCTTTAGTACACCT
CACACATATGAGATGCC,CACACATATAAGATGCC,GGCATCTCATATGTGTG,GGCATCTTATATGTGTG
CGAAGTGATGAAGGTGACAA,CGAAGTGATGAATGTGACAA,TTGTCACCTTCATCACTTCG,TTGTCACATTCATCACTTCG
ACAGAAATACAGTATATAACCT,CAGAAATACAGTAAATAACCT,AGGTTATATACTGTATTTCTGT,AGGTTATTTACTGTATTTCTG
CTGCCATGGAGGAAG,CTGCCATAGAGGAAG,CTTCCTCCATGGCAG,CTTCCTCTATGGCAG
TGATGTGAGAGCTCA,TGATGTGAAAGCTCA,TGAGCTCTCACATCA,TGAGCTTTCACATCA
CTCTCCCCACCCTCAG,CTCTCCCCATCCTCAG,CTGAGGGTGGGGAGAG,CTGAGGATGGGGAGAG
CCCCTACTTACCATGCACT,CCCCTACTTACCCTGCACT,AGTGCATGGTAAGTAGGGG,AGTGCAGGGTAAGTAGGGG
AAGCTCTTCAG,AAGCTTTTCAG,CTGAAGAGCTT,CTGAAAAGCTT
GACACCCGCTC,GACACTCGCTC,GAGCGGGTGTC,GAGCGAGTGTC
CGTTAATATTA,CGTTAGTATTA,TAATATTAACG,TAATACTAACG
ACTGCCTCAGC,ACTGCTTCAGC,GCTGAGGCAGT,GCTGAAGCAGT
TGGATACAACC,TGGATCCAACC,GGTTGTATCCA,GGTTGGATCCA
GAGGGAGAGAG,GAGGGGGAGAG,CTCTCTCCCTC,CTCTCCCCCTC
TCTCCCCCAGT,TCTCCTCCAGT,ACTGGGGGAGA,ACTGGAGGAGA
ACCATAGGGGT,ACCATGGGGGT,ACCCCTATGGT,ACCCCCATGGT
AAAATGACACA,AAAATTACACA,TGTGTCATTTT,TGTGTAATTTT
CACAC[GT]CCAGAC,CACAC[GT]TCAGAC,GTCTGGACGTGTG,GTCTGAACGTGTG
AGATTCGCAAC,AGATTTGCAAC,GTTGCGAATCT,GTTGCAAATCT
AGCCTGTAGAGCTGTTTATACAGTG,AGCCTTTAGAGCTGTTTATACAGTG,CACTGTATAAACAGCTCTACAGGCT,CACTGTATAAACAGCTCTAAAGGCT
CTTGCACACCT,CTTGCGCACCT,AGGTGTGCAAG,AGGTGCGCAAG
AATTCATCTGC,AATTCCTCTGC,GCAGATGAATT,GCAGAGGAATT
TTGTCCGCAAA,TTGTCTGCAAA,TTTGCGGACAA,TTTGCAGACAA
AGTGGTGAAGCCTC,TAGTGGTAAAGCCTC,GAGGCTTCACCACT,GAGGCTTTACCACTA
ATCAGGGCCGATCCA,CAGGGCAGATCCA,TGGATCGGCCCTGAT,TGGATCTGCCCTG
CTGCCTCCCGTCTGA,CTGCCTCCCATCTGA,TCAGACGGGAGGCAG,TCAGATGGGAGGCAG
TGTTGTAAGTTATTTAATTTATG,TTGTAAGTTATTTCATTTATG,CATAAATTAAATAACTTACAACA,CATAAATGAAATAACTTACAA
TGCTGGTGCCCCCGTG,TGCTGGTGCTCCCGTG,CACGGGGGCACCAGCA,CACGGGAGCACCAGCA
AAACAGAGATATGGTATGGC,AACAGAGATATGCTATGGC,GCCATACCATATCTCTGTTT,GCCATAGCATATCTCTGTT
CATGTAACTGATTGAACAAA,ATGTAACTGATCGAACAAA,TTTGTTCAATCAGTTACATG,TTTGTTCGATCAGTTACAT
CCCATCCAACAGGAGAG,CCCATCCAACATGAGAG,CTCTCCTGTTGGATGGG,CTCTCATGTTGGATGGG
TGTCCGACGGTTTTG,ATGTCCGACTGTTTTG,CAAAACCGTCGGACA,CAAAACAGTCGGACAT
ATCGAGCCAAAACTCGGGAT,ATCGAGCCGAAACTCGGGAT,ATCCCGAGTTTTGGCTCGAT,ATCCCGAGTTTCGGCTCGAT
CCTGACAGTAAGATATATCAA,CTGACAGTAAGATGTATCAA,TTGATATATCTTACTGTCAGG,TTGATACATCTTACTGTCAG
CCGCTAGCCTG,CCGCTGGCCTG,CAGGCTAGCGG,CAGGCCAGCGG
ATACTACACAG,ATACTTCACAG,CTGTGTAGTAT,CTGTGAAGTAT
ATAAAGCCCTT,ATAAATCCCTT,AAGGGCTTTAT,AAGGGATTTAT
ATCAAACTCAAGTGCCGGGCCAGGCCA,ATCAAGCTCAAGTGCCGGGCCAGGCCA,TGGCCTGGCCCGGCACTTGAGTTTGAT,TGGCCTGGCCCGGCACTTGAGCTTGAT
TCTCTCGTCTA,TCTCTTGTCTA,TAGACGAGAGA,TAGACAAGAGA
ATGGGTTCAAATGTGACCC,ATGGGTTCAAATCTGACCC,GGGTCACATTTGAACCCAT,GGGTCAGATTTGAACCCAT
AAGCTTGATTTCTTTTCTT,AAGCTTGATTTATTTTCTT,AAGAAAAGAAATCAAGCTT,AAGAAAATAAATCAAGCTT
CTCTCCTCGGTGTTGC,CTCTCCTCAGTGTTGC,GCAACACCGAGGAGAG,GCAACACTGAGGAGAG
CCTAAGTTAACAGAATTT,CTAAGTTACCAGAATTT,AAATTCTGTTAACTTAGG,AAATTCTGGTAACTTAG
TCGAGTCTGGAGTCTTGA,CGAGTCTGGACTCTTGA,TCAAGACTCCAGACTCGA,TCAAGAGTCCAGACTCG
CTGAAAATAAACGGTCTATTC,CTGAAAATAAACAGTCTATTC,GAATAGACCGTTTATTTTCAG,GAATAGACTGTTTATTTTCAG
CAGCTGTTTTTCTTGTAGCCT,AGCTGTTTTTCTTATAGCCT,AGGCTACAAGAAAAACAGCTG,AGGCTATAAGAAAAACAGCT
CGAGGAGGACCCCCAGACA,CGAGGAGGACCTCCAGACA,TGTCTGGGGGTCCTCCTCG,TGTCTGGAGGTCCTCCTCG
CAGAGTGAACTGTATGATC,CAGAGTGAACTCTATGATC,GATCATACAGTTCACTCTG,GATCATAGAGTTCACTCTG
CCTGTCTGCAATACACTAG,CTGTCTGCAATGCACTAG,CTAGTGTATTGCAGACAGG,CTAGTGCATTGCAGACAG
CCCTGTCATTCAAACAT,CCCTGTCATTAAAACAT,ATGTTTGAATGACAGGG,ATGTTTTAATGACAGGG
CCTGAAAACAGACAAACA,CCTGAAAACAGCCAAACA,TGTTTGTCTGTTTTCAGG,TGTTTGGCTGTTTTCAGG
TCGGAAGCGTC,TCGGAGGCGTC,GACGCTTCCGA,GACGCCTCCGA
AATCAAGGTCA,AATCACGGTCA,TGACCTTGATT,TGACCGTGATT
CAACACGGAGT,CAACATGGAGT,ACTCCGTGTTG,ACTCCATGTTG
GCTCTCCACTC,GCTCTTCACTC,GAGTGGAGAGC,GAGTGAAGAGC
TCAGAAACACA,TCAGACACACA,TGTGTTTCTGA,TGTGTGTCTGA
AAAGGGTTTTC,AAAGGTTTTTC,GAAAACCCTTT,GAAAAACCTTT
AGATTAAGGGG,AGATTGAGGGG,CCCCTTAATCT,CCCCTCAATCT
GATAAAACTGTGGGTTTGCACAACCAAAGAATTT,GATAAAACTGTGGGTTTGCACAACCAAATAATTT,AAATTCTTTGGTTGTGCAAACCCACAGTTTTATC,AAATTATTTGGTTGTGCAAACCCACAGTTTTATC
GACCAAGATAC,GACCAGGATAC,GTATCTTGGTC,GTATCCTGGTC
CCCGGGTCTCG,CCCGGTTCTCG,CGAGACCCGGG,CGAGAACCGGG
CCAGAAACAACTTAATTAT,CAGAAACAACTCAATTAT,ATAATTAAGTTGTTTCTGG,ATAATTGAGTTGTTTCTG
CCAGCTATAACAGCCTTG,CAGCTATAACAACCTTG,CAAGGCTGTTATAGCTGG,CAAGGTTGTTATAGCTG
CACGTTTTCTCTTTTCTC,ACGTTTTCTCCTTTCTC,GAGAAAAGAGAAAACGTG,GAGAAAGGAGAAAACGT
CAGGAACGGGTATGAG,CAGGAACGAGTATGAG,CTCATACCCGTTCCTG,CTCATACTCGTTCCTG
AGAAAAGGTGAGCTAGTATG,AAAAGGTGAGCTCGTATG,CATACTAGCTCACCTTTTCT,CATACGAGCTCACCTTTT
AGGTGTGAGATATTAATAAT,AGGTGTGAGATATTGATAAT,ATTATTAATATCTCACACCT,ATTATCAATATCTCACACCT
AGAGGTCTGTCCGTGATAA,AGAGGTCTGTCCATGATAA,TTATCACGGACAGACCTCT,TTATCATGGACAGACCTCT
CAGCGATGTGGACATG,CCAGCGATGTTGACATG,CATGTCCACATCGCTG,CATGTCAACATCGCTGG
CCGGGAAGATT,CCGGGGAGATT,AATCTTCCCGG,AATCTCCCCGG
TTGAA[CT]TACAGTTCAGCTGT,TTGAA[CT]TGCAGTTCAGCTGT,ACAGCTGAACTGTAAGTTCAA,ACAGCTGAACTGCAAGTTCAA
CTGCACGCCTC,CTGCATGCCTC,GAGGCGTGCAG,GAGGCATGCAG
AGAGCATG,AGGGCATG,CATGCTCT,CATGCCCT
AGGGACGGGCA,AGGGATGGGCA,TGCCCGTCCCT,TGCCCATCCCT
GTAGACGAGAG,GTAGATGAGAG,CTCTCGTCTAC,CTCTCATCTAC
ACAATAATTTT,ACAATTATTTT,AAAATTATTGT,AAAATAATTGT
ATACTGCTCCAGAGACG,CTGCTCCAGGGACG,CGTCTCTGGAGCAGTAT,CGTCCCTGGAGCAG
ATGACGAAGCTCTCC,ATGACGAAACTCTCC,GGAGAGCTTCGTCAT,GGAGAGTTTCGTCAT
CTGATCGATTTCAACCC,TGATCGATCTCAACCC,GGGTTGAAATCGATCAG,GGGTTGAGATCGATCA
CGTACGAGATGTAGATGT,CGTACGAGATATAGATGT,ACATCTACATCTCGTACG,ACATCTATATCTCGTACG
CAAGAGGTCTTTGTCGACAT,AAGAGGTCTTTGCCGACAT,ATGTCGACAAAGACCTCTTG,ATGTCGGCAAAGACCTCTT
AAACACAAAACCCC,AAACACAAACCCCC,GGGGTTTTGTGTTT,GGGGGTTTGTGTTT
ATGCGGGTATTTAGCCTTAA,CGGGTATTTCGCCTTAA,TTAAGGCTAAATACCCGCAT,TTAAGGCGAAATACCCG
TTGCCTCCCCCGCTC,TTATTGCCTCTCCCGCTC,GAGCGGGGGAGGCAA,GAGCGGGAGAGGCAATAA
ACATGAATGTCTTTTTTC,CATGAATGTATTTTTTC,GAAAAAAGACATTCATGT,GAAAAAATACATTCATG
ATCCGAAATCACAGCCAT,CGAAATCGCAGCCAT,ATGGCTGTGATTTCGGAT,ATGGCTGCGATTTCG
TGCCAATTCCGCCAGAGGGCGG,TGCCAATTCCACCAGAGGGCGG,CCGCCCTCTGGCGGAATTGGCA,CCGCCCTCTGGTGGAATTGGCA
TCTTTTTGTGTGTTTCTCTT,TTTTTGTGTGTGTCTCTT,AAGAGAAACACACAAAAAGA,AAGAGACACACACAAAAA
CATGAGTGTGATATGGG,CATGAGTGTGTTATGGG,CCCATATCACACTCATG,CCCATAACACACTCATG
CATTAATGCCA,CATTACTGCCA,TGGCATTAATG,TGGCAGTAATG
TACTTAATTTT,TACTTTATTTT,AAAATTAAGTA,AAAATAAAGTA
CACAGCT[GT]CAGAC,CACAGGT[GT]CAGAC,GTCTGACAGCTGTG,GTCTGACACCTGTG
TTCTGTGAGAGATTTAG,TTTCTGTGAGATATTTAG,CTAAATCTCTCACAGAA,CTAAATATCTCACAGAAA
CGTTGGTGTGAAGTACAGAT,TTGGTGTGAAGGACAGAT,ATCTGTACTTCACACCAACG,ATCTGTCCTTCACACCAA
CAATAAAGTGCTAGGTGTCC,CAATAAAGTGCTAAGTGTCC,GGACACCTAGCACTTTATTG,GGACACTTAGCACTTTATTG
TCTGGAGGCGTATTGG,CTGGAGGCATATTGG,CCAATACGCCTCCAGA,CCAATATGCCTCCAG
CATCAATACACGTTCTAAG,CATCAATACACATTCTAAG,CTTAGAACGTGTATTGATG,CTTAGAATGTGTATTGATG
ACTCTCACAGATTACGGTAC,CTCTCACAGATAACGGTAC,GTACCGTAATCTGTGAGAGT,GTACCGTTATCTGTGAGAG
CTGTGTGAATTGCCTC,CTGTGTGAACTGCCTC,GAGGCAATTCACACAG,GAGGCAGTTCACACAG
TTGTAACGGACCTTCCCCT,AACGGACCGTCCCCT,AGGGGAAGGTCCGTTACAA,AGGGGACGGTCCGTT
TCCAAGAGGCTGCGCC,TCCAAGAGGCTACGCC,GGCGCAGCCTCTTGGA,GGCGTAGCCTCTTGGA
CTACACACACTCTCACACAC,CACACACTCACACACAC,GTGTGTGAGAGTGTGTGTAG,GTGTGTGTGAGTGTGTG
TGCACATCAGTCGACCTG,CACATCAGGCGACCTG,CAGGTCGACTGATGTGCA,CAGGTCGCCTGATGTG
CACCTCCAGATTTAC,ACCTCCGGATTTAC,GTAAATCTGGAGGTG,GTAAATCCGGAGGT
AGCCTTCAGAAAGTATTC,CCTTCAGAACGTATTC,GAATACTTTCTGAAGGCT,GAATACGTTCTGAAGG
CTCTGCAACCTGTTCAG,TCTGCAACCCGTTCAG,CTGAACAGGTTGCAGAG,CTGAACGGGTTGCAGA
AGTATCTGCCC,AGTATTTGCCC,GGGCAGATACT,GGGCAAATACT
CAGCTCCTGGACTCA,CAGCTCCTAGACTCA,TGAGTCCAGGAGCTG,TGAGTCTAGGAGCTG
CAAACACACACAGAGCC,AACACACGCAGAGCC,GGCTCTGTGTGTGTTTG,GGCTCTGCGTGTGTT
TCTGGGCGTACTGCC,CTGGGCATACTGCC,GGCAGTACGCCCAGA,GGCAGTATGCCCAG
CATTCACACTGATCTGC,TTCACACAGATCTGC,GCAGATCAGTGTGAATG,GCAGATCTGTGTGAA
ATGGCGTAAACAA,ATGGCATAAACAA,TTGTTTACGCCAT,TTGTTTATGCCAT
CTCTGACGGTGTGCTT,TCTCTGACAGTGTGCTT,AAGCACACCGTCAGAG,AAGCACACTGTCAGAGA
CCTGTTTGTTGAAACCGTA,CCTGTTTGTTGTAACCGTA,TACGGTTTCAACAAACAGG,TACGGTTACAACAAACAGG
CATTGGCGGTGTAACG,TCATTGGCAGTGTAACG,CGTTACACCGCCAATG,CGTTACACTGCCAATGA
CCCCTGGGTTTGTGTAT,CCCTGGGTGTGTGTAT,ATACACAAACCCAGGGG,ATACACACACCCAGGG
CCATTAATGAG,CCATTGATGAG,CTCATTAATGG,CTCATCAATGG
AGGGGCTTGTGAAGGATGTTGTCATA,AGGGGTTTGTGAAGGATGTTGTCATA,TATGACAACATCCTTCACAAGCCCCT,TATGACAACATCCTTCACAAACCCCT
ATGCCCAATTACGTCAGCA,TGCCCAATTACATCAGCA,TGCTGACGTAATTGGGCAT,TGCTGATGTAATTGGGCA
ACAAACGGGCCAGTTG,CAAACGGGACAGTTG,CAACTGGCCCGTTTGT,CAACTGTCCCGTTTG
TTGTAAGTTGTGGCTAAAA,TTGTAAGTTGTGACTAAAA,TTTTAGCCACAACTTACAA,TTTTAGTCACAACTTACAA
ACGCATGCCCCTACC,ACGCATGTCCCTACC,GGTAGGGGCATGCGT,GGTAGGGACATGCGT
ACAGTGTCTTC,ACAGTTTCTTC,GAAGACACTGT,GAAGAAACTGT
AGTACAGCAGC,AGTACTGCAGC,GCTGCTGTACT,GCTGCAGTACT
ACCTAG,GCCTAG,CTAGGT,CTAGGC
AGCACATCTGG,AGCACGTCTGG,CCAGATGTGCT,CCAGACGTGCT
TCACCAAAAAG,TCACCCAAAAG,CTTTTTGGTGA,CTTTTGGGTGA
CTTAAGACTTT,CTTAATACTTT,AAAGTCTTAAG,AAAGTATTAAG
TAGTCTCTTTACATAATCTC,TAGTCTCTTTACTTAATCTC,GAGATTATGTAAAGAGACTA,GAGATTAAGTAAAGAGACTA
AAAGTTTGAAATGTGTTCTAC,AAAGTTTGAAATGTCTTCTAC,GTAGAACACATTTCAAACTTT,GTAGAAGACATTTCAAACTTT
CCCTCACGTTTCTCTTAA,CCTCACGTTACTCTTAA,TTAAGAGAAACGTGAGGG,TTAAGAGTAACGTGAGG
CTCCCATATAGGCGTAGTAG,CCCATATAGGCATAGTAG,CTACTACGCCTATATGGGAG,CTACTATGCCTATATGGG
AACTCCCGGTC,AACTCTCGGTC,GACCGGGAGTT,GACCGAGAGTT"

for i in $PROBES
do
cat /dfs/Omalley_Lab/dayan/chum_test_panel/genos/merged_hash.txt | ../software/seqkit grep -s -i -p $i | grep ">" | awk -F ";" '{sum += $3} END {print sum}' >> /dfs/Omalley_Lab/dayan/chum_test_panel/test_outputs/probe_counts.txt
done


#save this file as a script and submit with qsub
```

```{r, eval = FALSE}
#Woops, this didnt work for probes equences that use the [ACGT] fromat for degenerate bases, will just fill these in manually

#this includes

#Oke_RDDFW52567_35
#Oke_RDDFW69741_55
#Oke_RDDFW18411_78
#Oke_RDDFW95768_60

#lets make the probe inputs for these 4 markers and run one the probe counter script again

#Oke_RDDFW52567_35
p1 <- "AGTCTGAGGTGCGAAAGGAGCTCCTTAAACTTGACCCCTCTTTAAG"
p2 <- "AGTCTGAGGTGTGAAAGGAGCTCCTTAAACTTGACCCCTCTTTAAG"
rc(p1)
rc(p2)

#Oke_RDDFW69741_55 
p1 <- "CACACGCCAGAC"
p2 <- "CACACTTCAGAC"
rc(p1)
rc(p2)

#Oke_RDDFW18411_78 
p1 <- "TTGAACTACAGTTCAGCTGT"
p2 <- "TTGAATTACAGTTCAGCTGT"
rc(p1)
rc(p2)

Oke_RDDFW95768_60 CCCACACAGACTGAACACCA CACAGCT[GT]CAGAC CACAGGT[GT]CAGAC
#Oke_RDDFW95768_60
p1 <- "CACAGCTGCAGAC"
p2 <- "CACAGGTTCAGAC"
rc(p1)
rc(p2)
```


```{bash, eval = FALSE}
# now run the extra 4 markers

#!/bin/bash
#$ -S /bin/bash
#$ -N probe_count
#$ -cwd
#$ -o $JOB_NAME.out
#$ -e $JOB_NAME.err

PROBES="AGTCTGAGGTGCGAAAGGAGCTCCTTAAACTTGACCCCTCTTTAAG,AGTCTGAGGTGTGAAAGGAGCTCCTTAAACTTGACCCCTCTTTAAG,CTTAAAGAGGGGTCAAGTTTAAGGAGCTCCTTTCGCACCTCAGACT,CTTAAAGAGGGGTCAAGTTTAAGGAGCTCCTTTCACACCTCAGACT
CACACGCCAGAC,CACACTTCAGAC,GTCTGGCGTGTG,GTCTGAAGTGTG
TTGAACTACAGTTCAGCTGT,TTGAATTACAGTTCAGCTGT,ACAGCTGAACTGTAGTTCAA,ACAGCTGAACTGTAATTCAA
CACAGCTGCAGAC,CACAGGTTCAGAC,GTCTGCAGCTGTG,GTCTGAACCTGTG
"

for i in $PROBES
do
cat /dfs/Omalley_Lab/dayan/chum_test_panel/genos/merged_hash.txt | ../software/seqkit grep -s -i -p $i | grep ">" | awk -F ";" '{sum += $3} END {print sum}' >> /dfs/Omalley_Lab/dayan/chum_test_panel/test_outputs/probe_counts_broke.txt
done


#save this file as a script and submit with qsub
#integrate results into final results table manually in a text editor
```


After running these scripts combine the results in a text editor with marker names.

```{r, warning=FALSE, message=FALSE}
seq_test <- read_tsv("qc_results/seq_test3.txt")
colnames(seq_test) <- c("marker", "primer_count", "probe_count")
```

## Read Distribution

First, let's look at the distribution of reads.

```{r, warning=FALSE, message=FALSE}
# for this step it is best to look at reads containing the primer alone, as a read-hog primer can waste reads even if it amplifies off-target regions.
ggplot(data = seq_test)+geom_histogram(aes(x = primer_count))+theme_classic()
```

Looks like there are a handful of read hogs in there, let's examine those first.

Here we take an arbitrary cutoff (drawn from the read distribution above) of greater than 300,000 reads

```{r}
seq_test$portion_primer_reads <- seq_test$primer_count/sum(seq_test$primer_count)
kable(arrange(seq_test[seq_test$primer_count>3e5,], desc(primer_count)), digits = 3 )
```

We see here that the top four most observed primers (of 350) account for 25% of total reads. And the next top 14 compose an additional 19% of the reads. 

All but one of these demonstrate highly skewed primer/probe ratios, indicative of poor specificity for these primers. These are great candidates for removal or a change to the primer sequence to improve specificity.



__Failure to amplify__

While some low amplifying primers can be "rescued" by excluding read hogs from future runs, primers with zero or extremely low amplification may need to be redesigned, or the in silico sequence used may need to modified to accomodate unaccounted for variants in the primer region. Such markers might show low/zero primer counts, but large numbers of probe counts. Let's look at these.


Below we plot the read distribution overall (blue) (estimated by primer counts), vs reads from markers wiht less than 10% of the mean number of reads.

```{r, warning = FALSE, message = FALSE}
zero_primers <- seq_test %>%
  filter(primer_count < 0.1* mean(primer_count))

ggplot(data = zero_primers)+geom_histogram(aes(x = probe_count), fill = "red", alpha = 0.7)+geom_histogram(data = seq_test, aes(x = probe_count), fill = "blue", alpha = 0.5)+ theme_classic()
```

Nope, only 1 marker demonstrates less than 10% of the mean primer count. This is great news, it suggests all primers (but one) amplify well. Removing read hogs is unlikely rescue any markers, because very few amplify poorly


## Primer/Probe Specificity

First let's look at the distribution of primer and probe counts alone.

Primers first (ignoring the read hogs).
```{r, warning= FALSE, message = FALSE}
# for this step it is best to look at reads containing the primer alone, as a read-hog primer can waste reads even if it amplifies off-target regions.

ggplot(data = seq_test)+geom_histogram(aes(x =  primer_count))+theme_classic()+xlim(-1000, 300000)
```

Probes next.
```{r, warning= FALSE, message = FALSE}

ggplot(data = seq_test)+geom_histogram(aes(x =  probe_count),)+theme_classic()
ggplot(data = seq_test)+geom_histogram(aes(x =  probe_count),)+theme_classic()+xlim(-1000, 100000)+ggtitle("probe count\ntruncated at 100k")

```

__Ratios__

Now let's take a look at the ratio of those values. 
```{r, warning=FALSE, message=FALSE}
# for this step it is best to look at reads containing the primer alone, as a read-hog primer can waste reads even if it amplifies off-target regions.

ggplot(data = seq_test)+geom_histogram(aes(x = log10( primer_count/probe_count)))+theme_classic()+geom_vline(aes(xintercept = 0.69897), color = "blue")+geom_vline(aes(xintercept = -0.69897), color = "red")
```


Here we see that most primers and probes have sufficient specificity (i.e. primer:Probe ratio between 1:5 and 5:1). Very few markers demonstrate a large left skew in their primer:probe ratio (i.e. less than 1/5x, beyond the red line above). For these markers, there is likely insufficient specificity of the probe sequence, or variation in the region of the primer, such that amplification can still occur, but lack of a perfect match causes primer dropout in the seq_test script (and also in the GTseq_genotyper, as it is written to keep only exact matches).

In contrast many markers show rigt skew in the primer:probe ratio (i.e. greater 5x, blue line). This suggests that low primer specificity is leading to off target amplification, or probe region variation leading to allele dropout. Let's look more closely by seeing if this is skewed towards high amplifiers.

```{r, warning=FALSE, message=FALSE}
ggplot(data = seq_test)+geom_point(aes((probe_count), (primer_count)), alpha = 0.5)+geom_abline(aes(slope = 1, intercept = 0))+xlim(4,7)+ylim(4,7)+scale_x_log10()+scale_y_log10()+theme_classic()
```

Yes, it looks like most of the high primer:probe ratio markers (i.e. strong positive residuals from the 1:1 line in the plot above) occur among those with LOTs of primer counts, suggesting thee primers are amplifying off-target sequences, rather than bad probes/probe region variation.

```{bash, eval = FALSE}
# here we look at one marker with zero primer reads, but a lot of probe reads
# we search for sequences that match a probe sequence using exact match, then search the output for a partial match, of the primer
cat genos/merged_hash.txt | ../software/seqkit grep -s -i -p TCAGGGACGATAAAGGGATCATCTT | grep ">" | awk -F ";" '{sum += $3} END {print sum}'

#once again, the seq_test script is putting out the wrong values, let's run it again using awk and seq_test

```

# Genotyping

We also ran the genotyper to check for paralog sequences.


```{bash, eval=FALSE}
#!/bin/bash
#$ -S /bin/bash
#$ -t 1-95
#$ -tc 20
#$ -N GTseq-genotyperv3
#$ -cwd
#$ -o $JOB_NAME_$TASK_ID.out
#$ -e $JOB_NAME_$TASK_ID.err

export PERL5LIB='/home/fw/dayand/perl5/lib/perl5/x86_64-linux-thread-multi/'

FASTQS=(`ls -1 *fastq`)
INFILE=${FASTQS[$SGE_TASK_ID -1]}
OUTFILE=$(basename ${INFILE%.fastq}.genos)

GTSEQ_GENO="/dfs/Omalley_Lab/dayan/software/GTseq-Pipeline/GTseq_Genotyper_v3.1.pl
"

PROBE_SEQS="/dfs/Omalley_Lab/dayan/chum_test_panel/Oke_GTseq350_ProbeSeqs.csv"

perl $GTSEQ_GENO $PROBE_SEQS $INFILE > $OUTFILE

#save this code chunk as a file on the server and submit this with qsub -q harold scriptname from the directory you want the output .genos files
```


```{bash, eval = FALSE}
SGE_Batch -q harold -r compile -c 'perl /dfs/Omalley_Lab/dayan/software/GTseq-Pipeline/GTseq_GenoCompile_v3.pl > ../genotypes/Oke_GTs_0.1.csv'
```

Normally we would do some filtering at this step, but we want to look at ALL loci, so moving forward with no filtering.


```{r, warning=FALSE, message=FALSE}
gt <- read_csv("qc_results/Oke_GTs_0.1.csv")
marker_info <- read_tsv("qc_results/marker_info.txt")
marker_info$a1_count <- as.numeric(marker_info$a1_count)
marker_info$a2_count <- as.numeric(marker_info$a2_count)
marker_info$total_count <- marker_info$a1_count+marker_info$a2_count

```


## Genotyping Summary

```{r}
#marker_info %>%
#  summarise(mean=mean(total_count, na.rm = TRUE), median=median(total_count, na.rm = TRUE), sd=sd(total_count, na.rm = TRUE))
ggplot(data = marker_info)+geom_density(aes(x = (total_count)), fill = "#00204DFF", alpha = 0.5)+theme_classic()+geom_vline(aes(xintercept = median(marker_info$total_count, na.rm = TRUE)))+xlab("depth per locus per individual\nmedian depth = 272 (vertical line)")

```

Depth looks great to make GT calls. A subset of locusXindividual combos (i.e. individual GTs) have low depth, we will see if this due to bad individuals, genotypes or unaccounted for primer/probe region variants below.

## Paralogs

Here we plot allele counts for all loci.

First collect the allele count data from the .genos files.

```{bash, eval = FALSE}
touch marker_info.txt
for file in ../genos/*genos
do
    awk ' FS="," {print FILENAME,$1,$2,$3,$6,$7,$8}' $file >> marker_info.txt
done
```


Now we examine if any markers have a skewed allele count ratio for heterozygotes

```{r, warning = FALSE, message= FALSE}
library(lme4)
hets <- filter(marker_info, called_geno == "HET" | is.na(called_geno))

models <- hets %>%
  group_by(marker) %>%
  group_map(~ lm(a1_count ~ a2_count, data= .))

# Apply coef to each model and return a list of allele count ratios
lms <- lapply(models, coef)
ggplot()+geom_histogram(aes(x = sapply(lms,`[`,2)))+theme_classic()+ggtitle("allele ratios for all NA and HET calls")

#list of p-values
lms_anova <- lapply(models, summary)


# collect info about each bad model
paralog_possible <- which(abs(sapply(lms,`[`,2)) > 1.5) #bad because a positively skewed allele ratio
paralog_possible2 <- which(abs(sapply(lms,`[`,2)) < (2/3)) # bad because a negative skewed allele ratio

paralog_possible3 <- which(sapply(lms_anova, function(x) x$coefficients[,4][2])> 0.01) # bad because too much variance in allele ratio, even if mean ratio is 1

paralog_possible <- c(paralog_possible, paralog_possible2, paralog_possible3)
```

16 markers have skewed allele ratios (greater than 1.5), or high variance in allele ratios at heterozygote GTs, indicative of a possible paralogs.

Let's take a look at these.

```{r, warning=FALSE, message=FALSE}

plots <- marker_info %>%
  group_by(marker) %>%
  do(plots=ggplot(data=.)+geom_point(aes(a1_count, a2_count, color = called_geno))+theme_classic()+geom_abline(aes(slope=1, intercept=0))+geom_abline(aes(slope = 10, intercept=0), color = "green")+geom_abline(aes(slope = 0.1, intercept=0), color = "red")+geom_abline(aes(slope = 0.2, intercept=0), color = "blue")+geom_abline(aes(slope = 5, intercept=0), color = "blue")+coord_equal(ratio=1)+geom_abline(slope = -1, intercept = 10)+ggtitle(unique(.$marker)))


# we did this manually by iterating the command plots$plots[[paralog_possible[X]]] for x = 1:length(paralog_possible) and flagging any problematic markers, which we examine in detail later
```
Did not include all of the figures, in the notebook (see code chunk above if you want to look yourself), but most of these are simply artifacts of very few HET calls, so NAs from low read count individuals drive the ratio. However a few may be paralogs unaccounted for with current allele correction values, these are plotted below.

```{r, warning=FALSE, message=FALSE}

plots$plots[[paralog_possible[3]]]
plots$plots[[paralog_possible[6]]]
plots$plots[[paralog_possible[8]]]

```

Two of these three markers do not have enough putative heterozygotes to venture any guess as to what is going on here. However, Oke_U2001_629, looks like we may have an issue, but note that there is a strong allele correction value applied for the a1 allele here (2). If we were to run the genotype again WITHOUT the allele correction value, would this fix the problem?



```{r}
# undo the allele correction value
# the line from the genotyer script is "$Allele1_Count{$loci} = $Allele1_Count{$loci} - ($sum_xy / 4 * $A1_corr{$loci});"

#this implies the uncorrected allele_1 count is equal to (4F+C*A2)/(4-C), where C is the allele 1 correction value, A2 is the uncorrected allele2 count and F is the final allele 1 count (given by the genotyper)

#let's convert the corrected values
# NOTE this only works if the allele correction value for allele 2 = 0
marker_info$a1_uncorr <- ((4*marker_info$a1_count+marker_info$a1_corr*marker_info$a2_count)/(4-marker_info$a1_corr))

ggplot(data=filter(marker_info, marker == "Oke_U2001_629"))+geom_point(aes(a1_uncorr, a2_count, color = called_geno))+theme_classic()+geom_abline(aes(slope=1, intercept=0))+geom_abline(aes(slope = 10, intercept=0), color = "green")+geom_abline(aes(slope = 0.1, intercept=0), color = "red")+geom_abline(aes(slope = 0.2, intercept=0), color = "blue")+geom_abline(aes(slope = 5, intercept=0), color = "blue")+coord_equal(ratio=1)+geom_abline(slope = -1, intercept = 10)+ggtitle(unique("uncorrected Oke_U2001_269"))



```

Nope, that pushed it further the wrong way, perhaps we would need to increase allele correction value here. 

## Allele Dropout

Another QC check from the genotyped data is whether there are any individual GTs that have zero/very low reads when the marker and individual are otherwise good. This indicates that variation in either the primer or probe region of the amplicon is causing allele dropout or total marker dropout. This can be run using GTseq_GenoCompile_Counts.pl, but let's just do it in R (below)

```{r, warning=FALSE, message=FALSE}
marker_info$total_count <- marker_info$a1_count+marker_info$a2_count

geno_counts <- marker_info %>%
  select(total_count, marker, ind) %>%
  pivot_wider(id_cols = ind, names_from = marker, values_from = total_count)

# now lets normalize the read counts by the total depth at each individual to better identify locus level outliers
ind_depth <- rowSums(select(geno_counts, -ind))

normalized_geno_counts <- geno_counts %>% 
  select(-ind) %>% 
  mutate_all(~ ./ind_depth)*100

#hist(unlist(normalized_geno_counts[,-1]), breaks = 500, main = "normalized % total reads per locus per individual")
#hist(unlist(normalized_geno_counts[,-1]), xlim = c(0,1), breaks = 500, main = "normalized % total reads per locus per individual\nreadhogs excluded")

# but these zeros/low scores could also be from bad markers across ALL samples, let's normalize across columns too
normalized2 <- geno_counts <- as.data.frame(cbind(normalized_geno_counts[,1], (scale(normalized_geno_counts[,-1], center=FALSE, scale=colSums(normalized_geno_counts[,-1])))))

hist(table(which(normalized2 < 0.001, arr.ind = TRUE)[,2]), main= "number of individuals within a marker\ndemonstrating less than 0.1% normalized read depth")
hist(table(which(normalized2 == 0, arr.ind = TRUE)[,2]), main= "number of individuals within a marker\ndemonstrating 0 read depth")

# okay but now lets see if this just from extremely low coverage individuals
bad_depths <- ind_depth[which(normalized2 == 0, arr.ind = TRUE)[,1]]
ggplot()+geom_histogram(aes(x = ind_depth), color = "red", fill = "red", alpha = 0.2)+geom_histogram(aes(x = bad_depths), color = "blue", fill = "blue", alpha = 0.2)+theme_classic()+ggtitle("depths of individuals contributing to zero read GTs (blue), vs all depths (red)")

bad_depths2 <- ind_depth[which(normalized2 < 0.001, arr.ind = TRUE)[,1]]
ggplot()+geom_histogram(aes(x = ind_depth), color = "red", fill = "red", alpha = 0.2)+geom_histogram(aes(x = bad_depths2), color = "blue", fill = "blue", alpha = 0.2)+theme_classic()+ggtitle("depths of individuals contributing to low read read GTs (blue), vs all depths (red)")
```

While many markers (~170) contain individual GTs with abnormally low read depth (< 0.1% of normalized read depth), and many occur with zero read depth (~100), this never occurs across more than few individuals within a marker, and the vast majority of the occurences stem from the same individuals with so few reads that marker dropout might be occuring by chance alone. 

Taken together it looks like variants occuring the primer and probe sequence either do not occur or have sufficiently rare MAF that they unlikely to cause any major genotyping issues in these populations.

## Read Hogs Again

Another way to look at how bad the read hog problem is to compare to other GTseq runs without hashing their reads and counting primers. 

Here, we examine the distribution of on-target reads across loci for this panel, and for an unfiltered O mykiss panel using individuals from the Rogue river.

```{bash, eval = FALSE}
# grab %on-target for each locus for each individual (column 10 for each genos file)

for i in *genos
do
awk -F "," '{print $1, $10}' $i >> locus_ot_percent.txt
done

for i in *genos
do
awk -F "," '{print $1, $10}' $i >> locus_ot_percent_rogue_mykiss.txt
done


```

```{r, warning=FALSE, message=FALSE}
ot_perc <- read_delim("qc_results/locus_ot_percent.txt", delim = " ")
ot_perc_mykiss <- read_delim("qc_results/locus_ot_percent_rogue_mykiss.txt", delim = " ")


on_target_locus <- bind_rows(list(ot_perc=ot_perc, ot_perc_mykiss=ot_perc_mykiss), .id = 'source')
on_target_locus <- on_target_locus[,c(1:3)]
colnames(on_target_locus) <- c("species", "marker", "on_target_percent")
#this will have some NA rows from the header of each input genos, file, this is okay for now

ggplot(data = on_target_locus)+geom_density(aes(x = on_target_percent, fill = species), alpha = 0.5)+scale_fill_viridis_d(labels = c("O. keta", "O. mykiss\n(Rogue)"))+theme_classic()+xlab("percent of on-target reads per locus per individual")
```



```{r, message=FALSE, warning=FALSE}
omy_raw_gts <- read_csv("../../premature_migration/Rogue_Half-Pounder/half-pounder/genotype_data/half_pounder_GTs_0.1.csv")

gt_combined <- bind_rows(list(gt=gt, omy_raw_gts=omy_raw_gts), .id = 'source')

ggplot(data = gt_combined)+geom_density( aes(x = `%On-Target`, fill = source), alpha = 0.5)+scale_fill_viridis_d(labels = c("O. keta", "O. mykiss\n(Rogue)"))+theme_classic()+xlab("mean on target percentage per individual")

#ggplot()+geom_histogram(data = gt, aes(x = `%On-Target`), fill = "#00204DFF", alpha = 0.5)+geom_histogram(data = omy_raw_gts, aes(x = `%On-Target`), fill ="#FFEA46FF", alpha = 0.5)+theme_classic()
```

There are somewhat more off-target reads in the O. keta panel than O. mykiss, but this seems to be due to a handful of individuals with very low genotyping success overall, and probably not to markers performing poorly in a subset of individuals due to primer or probe sequence variation.

# Zero Read Marker

What's going on with the zero read marker? Oke_KPNA2_87, has 0 primer counts AND zero probe counts. Variant in both the primer and probe sequence OR Missing/bad primer in aliquot. 

To check we will do a fuzzy match (up to two mismatches) across the hash file with the primer sequence.

```{bash, eval = FALSE}
cat /dfs/Omalley_Lab/dayan/chum_test_panel/genos/merged_hash.txt | ../software/seqkit grep -s -i -p AGGCAGCCAGGTAAGTCAGTA -m 2 > KPNA2_87_2mismatch_hash.txt
```

Nope, 17 matching reads (out of 50 million) even allowing for two mismatches. Note that this search is just looking for SNPs, indels may also cause primer dropout. 

# GTseq_Primer-Interaction-Test 

DID NOT RUN THIS

__NOTE__ I don't have the exact reverse primer sequences of the aliquot we received, only fwd primer and probe sequences. However, there's no reason to expect that the primer-heterodimers should form to a greater or lesser extent just because we are using the primers in a different population. Let's just trust the WDFW optimization here. 


# Summary / Reccomendation

Overall, this looks like a well optimized panel, and there is little reason to believe there will be any major problems using it with Oregon Chum populations despite being developed using data from WA populations. 

__Primer Specificity__  

The biggest issue with the panel is low primer specificity at a handful of markers, leading to high off-target amplification and wasting of reads. The top 10% markers compose 57% of all reads. The top 4 (1%) compose 25% of total reads. Since this level of over-amplification is not too severe relative to what we see with the O. mykiss panel, I'd recommend simply lowering the the primer concentration in the mix for the most overamplifying markers, rather than redesigning primers. 

If this is not an option (for example, if we receive an aliquot of mixed primers from WDFW), then I suspect the panel will operate fine as long as we do not push the sample size per lane to an extreme. Median read depth (including some poorly sequencing individuals) was 272 with ~ 500k reads allocated per individual in the lane (i.e. before filtering). This suggests we could get similar genotyping performance out of a full lane (assuming 350 million reads) with 700 individuals.


__Variation in OR Population__  

I did not observe any strong evidence that variation in OR populations will lead to marker or allele dropout using these primers and probes and the GTseq pipeline. There were some individual gentoypes with abnormally low genotyping success within a marker that otherwise sequenced well, suggesting that variation may be leading to marker dropout, but these gentoypes were almost always restricted to individuals with poor overall coverage, suggesting that marker dropout is more likely due to chance than variation.

Furthermore, there's little reason to believe variation between WA and OR populations will lead to issues with paralagous variation. Allele correction values provided by WDFW work well for all but 1 marker in our genotyping run.

__Note of caution__  

My only note of caution here is that a substantial portion of the test panel individuals sequenced very poorly. Specifically, 25 (of 95) individuals had less than 10% on target reads, despite a high number of reads containing their barcode. I don't have a good explanation for this high number of poorly genotyped individuals, but I don't have any reason to think this is due to variation stemming from moving to a new region (OR vs WA), because these individuals sequence poorly at all markers.
